### DISCUSSION TITLE
Embedding in SharePoint

### DISCUSSION THREAD

**Discussion by jglassman2** [microsoft/sample-app-aoai-chatGPT#661] [Q&A]
I'm running into a strange issue when embedding this in SharePoint. Everything will work correctly, but when the chat generates new responses, the SharePoint page will scroll down to the bottom of the page, usually pushing the chat offscreen. I'm not sure if this is an issue with how SharePoint embeds work or if something in this app is causing the scroll issues. I've posted on the SharePoint developer forum in case the issue is on that end, but no one is responding, so I figured I might as well ask here as well, in case there is something I can change in the app so it won't scroll automatically as responses are generated. Has anyone else run into a similar issue when embedding?

**Comment by borgerta** [comment]
Ran into the exact same issue.  Any luck resolving it?

**Comment by borgerta** [comment]
I've found a solution to this issue by disabling the auto-scroll in the app.

Go to /frontend/src/pages/chat/Chat.tsx

comment out the following code:

```
useLayoutEffect(() => {
        chatMessageStreamEnd.current?.scrollIntoView({ behavior: "smooth" })
    }, [showLoadingMessage, processMessages]);
```

you will need to manually scroll down in the embedded page as new content is generated, but the parent SharePoint page will no longer scroll down on it's own.

**Comment by LucaPascucci** [comment]
@jglassman2 

Can i ask you the step to embed this solution to sharepoint? I'm trying to create a web part for sharepoint but i'm not find a valid walkthrough to create it 😄 

Thanks
